### DISCUSSION TITLE
SIGSEGV in vips_image_guess_interpretation

### DISCUSSION THREAD

**Discussion by Keruspe** [libvips/libvips#2147] [Q&A]
When building libvips with optimisations enabled (-O2 -march=native), I hit a segmentation fault:

```
Thread 4 "environment" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fffcbb4c640 (LWP 663683)]
0x00007ffff7c4bcb4 in vips_image_guess_interpretation (image=0x7fffc0080000) at header.c:595
595     header.c: No such file or directory.
```

https://github.com/libvips/libvips/blob/460efa73fa00f90c09aa9c63c322f0dee1c60b10/libvips/iofuncs/header.c#L595

Not sure what could be causing this. I'm currently in the process of debugging third-party code which leads to this.
Without optimisations, I get this, if it can help:

```
GLib-GObject-WARNING **: 15:21:24.809: value "((VipsInterpretation) -1431655766)" of type 'VipsInterpretation' is invalid or out of range for property 'space' of type 'VipsInterpretation'
```

**Comment by jcupitt** [comment]
Hi @Keruspe,

That's a strange one.  At a guess, perhaps you have too many unrefs and the image has been freed while still in use? 

You'd need to make a test case somehow.

**Comment by Keruspe** [comment]
The same program works fine on other systems, which makes it even weirder

**Comment by jcupitt** [comment]
That sounds like a double free to me. I'd look for `object->ref_count` going below zero, then try counting the unrefs in your code.

**Comment by Keruspe** [comment]
I do not know the codebase at all as it's provided by a third party, but crossing your comment with some stack trace, I was able to find this fix, which seems to solve my problem (waiting for proper confirmation): https://github.com/h2non/bimg/pull/367

**Comment by jcupitt** [comment]
LGTM.
